### ISSUE TITLE
using np.array circumvents ambiguity error for offset units

### ISSUE THREAD

**Issue by mmarras** [hgrecco/pint#1335]
pint: 0.17
numpy: 1.20.3
python: 3.9.1
________
I observe the following:
```
ureg.autoconvert_offset_to_baseunit
```
\>>>False
```
myoven = (400 * ureg.K).to('degC')
```
\>>>126.85000000000002 degree_Celsius

```
2* myoven 
```
\>>>OffsetUnitCalculusError: Ambiguous operation with offset unit (degree_Celsius, ). See https://pint.readthedocs.io/en/latest/nonmult.html for guidance.
```
np.array(2)*myoven
```
\>>>253.70000000000005 degree_Celsius

Not sure if is a bug or intended, but it is somewhat inconsistent imho.

**Comment by jules-ch** [comment]
Yeah no check is done for OffsetUnit when delegating to Array protocol for pint

https://github.com/hgrecco/pint/blob/1d2891ca50a2c40221050eb29260d0d32ebfb85a/pint/numpy_func.py#L176
